doozer/doozerlib/backend/rebaser.py

```diff
@@ -872,31 +872,57 @@ def _add_build_repos(self, dfp: DockerfileParser, metadata: ImageMetadata, dest_
 
             # The value we will set REMOTE_SOURCES_DIR to.
             remote_source_dir_env = '/tmp/cachito-emulation'
-
-            for npm_entry in metadata.config.cachito.packages.npm:
-                # cachito creates an .npmrc file in the "npm" package manager paths. This
-                # used to cause NPM to use an NPM server, supplied by cachito,
-                # which only supplied the component's NPM deps.
-                # See https://github.com/containerbuildsystem/cachito/blob/fe2ef761d7d8169597166e9f92ee8ef3d62a091c/README.md .
-                # Dockerfiles may reference this file. We create it in emulation mode, but do
-                # not populate it. cachi2 works fundamentally differently
-                # by re-writing package-lock.json to reference local files.
-                # https://github.com/hermetoproject/cachi2/blob/2235854bfc5cb76ea168cb04d50be16efce9fa0a/docs/npm.md
-                if 'path' in npm_entry:
+            pkg_managers = metadata.config.content.source.pkg_managers
+
+            if "npm" in pkg_managers:
+                flag = False
+                for npm_entry in metadata.config.cachito.packages.npm:
+                    # cachito creates an .npmrc file in the "npm" package manager paths. This
+                    # used to cause NPM to use an NPM server, supplied by cachito,
+                    # which only supplied the component's NPM deps.
+                    # See https://github.com/containerbuildsystem/cachito/blob/fe2ef761d7d8169597166e9f92ee8ef3d62a091c/README.md .
+                    # Dockerfiles may reference this file. We create it in emulation mode, but do
+                    # not populate it. cachi2 works fundamentally differently
+                    # by re-writing package-lock.json to reference local files.
+                    # https://github.com/hermetoproject/cachi2/blob/2235854bfc5cb76ea168cb04d50be16efce9fa0a/docs/npm.md
+                    if 'path' in npm_entry:
```

Comment:
If `path` doesn't exist, it should default to `.`, then the .npmrc file located at the top directory of the source code. So a proper implementation could be:

```python
app_path = gomod_deps_path.joinpath('app').joinpath(npm_entry.get('path', '.'))
```

Reply:
The issue is that, in some cases, in image config, metadata.config.content.source.pkg_managers will be populated but metadata.config.cachito.packages.npm will be empty. In that case, we assume that the path is the current directory